script/daylight_factor.py

```diff
@@ -0,0 +1,92 @@
+"""
+A sample code to show case how one can write scripts to run studies using this folder
+structure.
+
+Usage:
+
+daylight_factor -f [folder]
+
+TODO:
+    - add functionality for several grid points.
+    - add radiance parameters file (most likely to a config file)
+"""
+import argparse
+import os
+import yaml
+import subprocess
+
+SUBFOLDERS = [
+    '0-model/etc/opaque', '0-model/etc/opaque/indoor', '0-model/etc/opaque/outdoor',
+    '0-model/etc/nonopaque', '0-model/etc/nonopaque/indoor',
+    '0-model/etc/nonopaque/outdoor'
+]
+
+def find_files(parent_folder, subfolder):
+    """Find rad and mat files in a target folder."""
+    files = []
+    folder = os.path.join(parent_folder, subfolder)
+    all_files = os.listdir(folder)
+    for fi in all_files:
+        if not fi.endswith('.rad'):
+            continue
+        fn, _ = os.path.splitext(fi)
+        # ensure a mat file with the same filename exist
+        assert os.path.isfile(os.path.join(folder, '%s.mat' % fn)), \
+            'Failed to find materil file for {}/{}'.format(subfolder, fi)
+        
+        files.extend(
+            [
+                os.path.normpath(os.path.join(subfolder, fn + '.mat')),
+                os.path.normpath(os.path.join(subfolder, fn + '.rad'))
+            ]
+        )
+    return files
+
+def main(args):
+    """ Run daylight factor."""
+    root_folder = args['folder'].strip()
+
+    scene_files = [f for folder in SUBFOLDERS for f in find_files(root_folder, folder)]
+    static_aperture_files = find_files(root_folder, r'0-model/aperture/static')
+    
+    state_file = os.path.join(root_folder, '0-model/aperture/dynamic', 'state.yaml')
+    with open(state_file) as inf:
+        states = yaml.load(inf)
+
+    # for now just run it for the first state and there is only one window group!
+    for window_group in states:
```

Comment:
Quick question about this structure of window_group > state. How are we planning on splitting options here? Are we going to run an entire simulation per window group and per state? Are we planning on keeping track of which grid of points corresponds to which window_group?

Reply:
eventually the second one but how to map grids to groups that they are related to but that is not an easy task to do. See this: https://github.com/ladybug-tools/honeybee/issues/158

In any case we will split the case for each window group at each state and that can run in parallel.

Reply:
Cool cool. We might have to think of a way of segregating certain points from being simulated by certain window groups in that case. eg: do 1 day with 4-5 bounces to get an idea of which points are affected by said window group. The idea being that we don't want to model the 5th floor window facing rooms if we're running a window group from 4th floor.